src/Components/ADT3DSceneBuilder/Internal/Behaviors/VisualRules/VisualRuleList.tsx

```diff
@@ -35,6 +35,23 @@ export const VisualRulesList: React.FC<IVisualRulesListProps> = ({
     );
 };
 
+function getBadgesAndMeshesCount(item: IVisualRule, t: TFunction<string>) {
+    let badgeCount = 0;
+    let meshColoringCount = 0;
+    item.conditions.forEach((condition) => {
+        if (condition.visual.iconName) {
```

Comment:
Nope, it would require a new schema change to add action type. I believe it is not necessary since we can determine the type if it contains or not an iconName

Reply:
maybe we can just pull this check out to the ViewerConfigUtility like we do for all the other types so this check is centralized at least. In the future, it might be worth putting a type on any new object we create where not all props are used for all variants of that object.